components/server/src/workspace/workspace-factory.ts

```diff
@@ -32,8 +36,8 @@ export class WorkspaceFactory {
     @inject(TracedWorkspaceDB) protected readonly db: DBWithTracing<WorkspaceDB>;
     @inject(ProjectDB) protected readonly projectDB: ProjectDB;
     @inject(TeamDB) protected readonly teamDB: TeamDB;
-    @inject(ConfigProvider) protected configProvider: ConfigProvider;
-    @inject(ImageSourceProvider) protected imageSourceProvider: ImageSourceProvider;
+    @inject(ConfigProvider) public configProvider: ConfigProvider;
```

Comment:
Why would this be necessary? :thinking:  Maybe it helps to extract `isGoodBaseforIncrementalPrebuild` as sueggested above.